File: app/src/main/java/me/yifeiyuan/flapdev/components/simpleimage/SimpleImageComponent.java
package me.yifeiyuan.flapdev.components.simpleimage;

import androidx.annotation.NonNull;
import android.view.View;

import me.yifeiyuan.flap.FlapComponent;
import me.yifeiyuan.flap.annotations.Component;
import me.yifeiyuan.flapdev.R;

/**
 * Created by 程序亦非猿 on 2018/12/4.
 */
@Component(layoutId = R.layout.flap_item_simple_image)
public class SimpleImageComponent extends FlapComponent<SimpleImageModel> {

    public SimpleImageComponent(final View itemView) {
        super(itemView);
    }

    @Override
    protected void onBind(@NonNull final SimpleImageModel model) {

    }
}

File: app/src/main/java/me/yifeiyuan/flapdev/components/simpleimage/SimpleImageComponent.kt
package me.yifeiyuan.flapdev.components.simpleimage

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import me.yifeiyuan.flap.delegate.AdapterDelegate
import me.yifeiyuan.flap.Component
import me.yifeiyuan.flap.annotations.Delegate
import me.yifeiyuan.flapdev.R
import me.yifeiyuan.flapdev.components.generictest.GenericFlapComponentDelegate

/**
 * Created by 程序亦非猿 on 2018/12/4.
 */
//@Delegate(layoutId = R.layout.flap_item_simple_image)
@Delegate(layoutName = "flap_item_simple_image")
class SimpleImageComponent(itemView: View) : Component<SimpleImageModel>(itemView) {
    override fun onBind(model: SimpleImageModel) {
    }
}

class SimpleImageComponentDelegate : AdapterDelegate<SimpleImageModel, SimpleImageComponent> {

    override fun delegate(model: Any): Boolean {
        return SimpleImageModel::class.java == model.javaClass
    }

    override fun onCreateViewHolder(inflater: LayoutInflater, parent: ViewGroup, viewType: Int): SimpleImageComponent {
        return SimpleImageComponent(inflater.inflate(R.layout.flap_item_simple_image, parent, false))
    }

}